geopandas/tests/test_dissolve.py

```diff
@@ -20,23 +20,24 @@ def nybb_polydf():
     nybb_polydf = nybb_polydf[["geometry", "BoroName", "BoroCode"]]
     nybb_polydf = nybb_polydf.rename(columns={"geometry": "myshapes"})
     nybb_polydf = nybb_polydf.set_geometry("myshapes")
-    nybb_polydf["manhattan_bronx"] = 5
-    nybb_polydf.loc[3:4, "manhattan_bronx"] = 6
+    nybb_polydf["island"] = [0, 2, 2, 1, 3]
```

Comment:
Is there a reason for those test changes? 

(in general, it might be better to have the test changes separately, because if the code changes are really only meant to improve performance, the tests should pass as is. But because the tests also changes it makes it harder to verify the perf improvement didn't actually accidentally changes something)

Reply:
To make them more thorough, following the issue raised by Martin in [this comment](https://github.com/geopandas/geopandas/pull/2102#issuecomment-1226920926). But you're right, this belongs to another PR, I reverted the commit.

Reply:
I would actually like to keep this test but you can adapt the dataframe within the test itself rather than changing it for all the tests. It is a case that was not covered as shown in that comment you link.

Reply:
Ok, done in [e9dafe9](https://github.com/geopandas/geopandas/pull/2102/commits/e9dafe9e56da136bae35879bf49f3c18a2fbcb66)